### PR TITLE
Entity Tameable - NEW

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR adds a new expression to get/set the owner/tamer of an entity (such as a horse or wolf) as well as a new condition to check if an entity is tameable.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	
+	@Override
+	protected OfflinePlayer[] get(Event e, LivingEntity[] source) {
+		return get(source, new Getter<OfflinePlayer, LivingEntity>() {
+			@Nullable
+			@Override
+			public OfflinePlayer get(LivingEntity entity) {
+				if (entity instanceof Tameable && ((Tameable) entity).isTamed()) {
+					return ((OfflinePlayer) ((Tameable) entity).getOwner());
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: To reduce casting overhead, perhaps you can consider doing
```
if (entity instanceof Tameable) {
	Tameable t = ((Tameable) entity);
	if (t.isTabled()){
		return (OfflinePlayer) t.getOwner();
	}
}
```

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: good thinking.

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	@Override
+	protected OfflinePlayer[] get(Event e, LivingEntity[] source) {
+		return get(source, new Getter<OfflinePlayer, LivingEntity>() {
+			@Nullable
+			@Override
+			public OfflinePlayer get(LivingEntity entity) {
+				if (entity instanceof Tameable) {
+					Tameable t = ((Tameable) entity);
+					if (t.isTamed())
+					return (OfflinePlayer) t.getOwner();
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Indentation is missing :P

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: sweet Jesus, how the heck did I miss that! Time for more coffee, haha! Thanks

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+@Since("INSERT VERSION")
+public class ExprEntityTamer extends PropertyExpression<LivingEntity, OfflinePlayer> {
+	
+	static {
+		register(ExprEntityTamer.class, OfflinePlayer.class, "(owner|tamer)", "livingentities");
+	}
+	
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Remove the final keyword on these.

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Entity Owner")
+@Description("The owner of a tameable entity, such as a horse or wolf.")
+@Examples({"set owner of target entity to player",
+	"delete owner of target entity",
+	"set {_t} to uuid of tamer of target entity"})
+@Since("INSERT VERSION")
+public class ExprEntityTamer extends PropertyExpression<LivingEntity, OfflinePlayer> {
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Why is this not a SimplePropertyExpression?

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+				}
+				return null;
+			}
+		});
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: May as well allow RESET?

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java (hunk 3 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: removed

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java (hunk 4 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Changed

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java (hunk 5 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Added

File: src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+public class ExprEntityTamer extends SimplePropertyExpression<LivingEntity, OfflinePlayer> {
+	
+	static {
+		register(ExprEntityTamer.class, OfflinePlayer.class, "(owner|tamer)", "livingentities");
+	}
+	
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+		setExpr((Expression<LivingEntity>) exprs[0]);
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Welp, now that it is a SPE, there is no point on overriding init.

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprEntityTamer.java]: Sorry about that, its been removed. I still get quite confused about all the different expression types. 
Im going to have to make myself a list of all of them, how they are used, and how exactly they are different, and their proper purposes.

**Pull request merged**